src/Rules/Library/ButtonInvokeAndTogglePatterns.cs

```diff
@@ -16,13 +16,14 @@ public ButtonInvokeAndTogglePatterns()
             this.Info.Description = Descriptions.ButtonInvokeAndTogglePatterns;
             this.Info.HowToFix = HowToFix.ButtonInvokeAndTogglePatterns;
             this.Info.Standard = A11yCriteriaId.NameRoleValue;
+            this.Info.ErrorCode = EvaluationCode.Error;
         }
 
-        public override EvaluationCode Evaluate(IA11yElement e)
+        public override bool PassesTest(IA11yElement e)
         {
-            var rule = Relationships.All(Patterns.Invoke, Patterns.Toggle);
+            var rule = ~(Relationships.All(Patterns.Invoke, Patterns.Toggle));
 
-            return rule.Matches(e) ? EvaluationCode.Error : EvaluationCode.Pass;
```

Comment:
Would it be any more efficient to leave rule unchanged, then negate the return value from Matches? That saves us creating a Condition object, and I think you've followed this pattern in previous changes

Reply:
I did that originally, and then changed it to this way. Honestly, I'm not crazy about the '!' negation, even in the cases I've used it. I think it makes it harder to easily groc the intent of the method.